client/src/commands/run.ts

```diff
@@ -156,6 +156,8 @@ async function runCode(selected?: boolean, uri?: Uri) {
     (_progress, cancellationToken) => {
       cancellationToken.onCancellationRequested(() => {
         session.cancel?.();
+        running = false;
+        commands.executeCommand("setContext", "SAS.running", false);
```

Comment:
It's done at the `finally` call at line 182 below. Is that the `runCode` promise never fulfilled if cancelled? I think it would be great if it can still be fulfilled even cancelled. That will make upper layer easier.

Reply:
Yeah, it's never fulfilled in that case. I half expected I could do something like `const token = cancellationToken.onCancellationRequested...` and call `token.dispose()` to attempt to resolve things, but that also didn't resolve things

Reply:
The cancellation token is from VS Code UI, the promise is from our own session, they're different.
Perhaps echo something to powsershell when cancel and capture it similar to -submit-end--

Reply:
Good idea. Done in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/bf8ac6a0a1babd95b1c6683b8fbd7ec17c0b9e06